langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AzureAiSearchEmbeddingStore.java

```diff
@@ -0,0 +1,426 @@
+                .setKey(true)
+                .setFilterable(true));
+        fields.add(new SearchField(DEFAULT_FIELD_CONTENT, SearchFieldDataType.STRING)
+                .setSearchable(true)
+                .setFilterable(true));
+        fields.add(new SearchField(DEFAULT_FIELD_CONTENT_VECTOR, SearchFieldDataType.collection(SearchFieldDataType.SINGLE))
+                .setSearchable(true)
+                .setVectorSearchDimensions(embeddingDimensions)
+                .setVectorSearchProfileName("vector-search-profile"));
+        fields.add((new SearchField(DEFAULT_FIELD_METADATA, SearchFieldDataType.COMPLEX)).setFields(
+                Arrays.asList(
+                        new SearchField(DEFAULT_FIELD_METADATA_SOURCE, SearchFieldDataType.STRING)
+                                .setFilterable(true),
+                        (new SearchField(DEFAULT_FIELD_METADATA_ATTRS, SearchFieldDataType.collection(SearchFieldDataType.COMPLEX))).setFields(
+                                Arrays.asList(
+                                        new SearchField("key", SearchFieldDataType.STRING)
+                                                .setFilterable(true),
+                                        new SearchField("value", SearchFieldDataType.STRING)
+                                                .setFilterable(true)
+                                )
+                        )
+
+                )
+        ));
+
+        VectorSearch vectorSearch = new VectorSearch()
+                .setAlgorithms(Collections.singletonList(
+                        new HnswAlgorithmConfiguration("vector-search-algorithm")
+                                .setParameters(
+                                        new HnswParameters()
+                                                .setMetric(VectorSearchAlgorithmMetric.COSINE)
+                                                .setM(4)
+                                                .setEfSearch(500)
+                                                .setEfConstruction(400))))
+                .setProfiles(Collections.singletonList(
+                        new VectorSearchProfile("vector-search-profile", "vector-search-algorithm")));
+
+        SemanticSearch semanticSearch = new SemanticSearch().setDefaultConfigurationName("semantic-search-config")
+                .setConfigurations(Arrays.asList(
+                        new SemanticConfiguration("semantic-search-config",
+                                new SemanticPrioritizedFields()
+                                        .setContentFields(new SemanticField(DEFAULT_FIELD_CONTENT))
+                                        .setKeywordsFields(new SemanticField(DEFAULT_FIELD_CONTENT)))));
+
+
+        SearchIndex index = new SearchIndex(INDEX_NAME)
+                .setFields(fields)
+                .setVectorSearch(vectorSearch)
+                .setSemanticSearch(semanticSearch);
+
+        searchIndexClient.createOrUpdateIndex(index);
```

Comment:
this config has a lot of parameters to tweak. should we maybe accept a `SearchIndex` as an optional param in ctor/builder? in case user wants more flexibility

Reply:
You are right, it's done!